File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> traversal = new ArrayList<>();
        List<TreeNode> stack = new ArrayList<>();
        TreeNode current = root;

        while (current != null || stack.size() > 0){
            while (current != null){
                stack.add(current);
                current = current.left;
            }
            current = stack.remove(stack.size() - 1);
            traversal.add(current.val);
            current = current.right;
        }
        return traversal;
    }
}

File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root: return
        
        dummy = TreeNode(root.val, root.right, root.left)
        # print(dummy.val, dummy.left.val)
        stack = []
        if dummy.left: stack.append(dummy.left)
        if dummy.right: stack.append(dummy.right)

        # stack = [3, 2, ]
        while stack:
            pop = stack.pop()
            if pop.left: stack.append(pop.left)
            if pop.right: stack.append(pop.right)
            
            pop.left, pop.right = pop.right, pop.left


        return dummy
        